packages/client/test/rpc/debug/storageRangeAt.spec.ts

```diff
@@ -0,0 +1,480 @@
+   */
+  interface TestSetup {
+    /**
+     * The server object to query for the tests.
+     */
+    server: HttpServer
+    /**
+     * The block that contains the transactions to be tested.
+     */
+    block: Block
+    /**
+     * The address of the created dummy contract.
+     */
+    createdAddress: Address
+    /**
+     * The address of another created dummy contract that does not contain storage.
+     */
+    createdAddressNoStorage: Address
+  }
+
+  beforeEach<TestSetup>(async (context) => {
+    // Populate a chain with three transactions: the first one deploys a contract,
+    // the second one updates a value in that contract, and the third one deploys
+    // another contract that does not put anything in its storage.
+
+    const { chain, common, execution, server } = await setupChain(genesisJSON, 'post-merge', {
+      txLookupLimit: 0,
+    })
+
+    const firstTx = TransactionFactory.fromTxData(
+      {
+        type: 0x2,
+        gasLimit: 10000000,
+        maxFeePerGas: 1000000000,
+        maxPriorityFeePerGas: 1,
+        value: 0,
+        data: storageBytecode,
+      },
+      { common, freeze: false }
+    ).sign(dummy.privKey)
+
+    const vmCopy = await execution.vm.shallowCopy()
+    const parentBlock = await chain.getCanonicalHeadBlock()
+    const blockBuilder = await vmCopy.buildBlock({
+      parentBlock,
+      headerData: {
+        timestamp: parentBlock.header.timestamp + BigInt(1),
+      },
+      blockOpts: {
+        calcDifficultyFromHeader: parentBlock.header,
+        putBlockIntoBlockchain: false,
+      },
+    })
+
+    const result = await blockBuilder.addTransaction(firstTx, { skipHardForkValidation: true })
+
+    const secondTx = TransactionFactory.fromTxData(
+      {
+        to: result.createdAddress,
+        type: 0x2,
+        gasLimit: 10000000,
+        maxFeePerGas: 1000000000,
+        maxPriorityFeePerGas: 1,
+        value: 0,
+        nonce: 1,
+        data: updateBytecode,
+      },
+      { common, freeze: false }
+    ).sign(dummy.privKey)
+
+    await blockBuilder.addTransaction(secondTx, { skipHardForkValidation: true })
+
+    const thirdTx = TransactionFactory.fromTxData(
+      {
+        type: 0x2,
+        gasLimit: 10000000,
+        maxFeePerGas: 1000000000,
+        maxPriorityFeePerGas: 1,
+        value: 0,
+        nonce: 2,
+        data: noStorageBytecode,
+      },
+      { common, freeze: false }
+    ).sign(dummy.privKey)
+
+    const thirdResult = await blockBuilder.addTransaction(thirdTx, { skipHardForkValidation: true })
+
+    const block = await blockBuilder.build()
+    await chain.putBlocks([block], true)
+
+    context.server = server
+    context.block = await chain.getCanonicalHeadBlock()
+    context.createdAddress = result.createdAddress!!
+    context.createdAddressNoStorage = thirdResult.createdAddress!!
+  })
+
+  it<TestSetup>('Should return the correct (number of) key value pairs.', async ({
```

Comment:
What does the `<TestSetup>` type parameter do?  is it just a label, or does it change the `it` function?

Reply:
It is a typesafe way of passing variables from `beforeEach()` to tests (see [here](https://vitest.dev/guide/test-context.html#typescript-1)). For legibility, I had already annotated the `TestSetup` interface.